File: Record.php
<?php

namespace Samples\Ignite\Php;

class Record
{
    public $id;
    public $price;
    public $quantity;

    public function __construct() {}

    public static function create(string $id, float $price, int $quantity): Record
    {
        $r = new Record();

        $r->id = $id;
        $r->price = $price;
        $r->quantity = $quantity;

        return $r;
    }

    public function __toString() {
        return '{id: \'' . $this->id . '\' price: ' . $this->price . ' quantity: ' . $this->quantity . '}';
    }
}

File: Samples.Ignite.Php/Record.cs
namespace Samples.Ignite.Php
{
    public class Record
    {
        // Ignite usability issue: must keep the field names same as PHP's - all camelCase.
        public string id { get; set; }

        public double price { get; set; }

        public int quantity { get; set; }

        public override string ToString()
        {
            return $"{{{nameof(id)}: '{id}', {nameof(price)}: {price}, {nameof(quantity)}: {quantity}}}";
        }
    }
}
